keras_rs/src/layers/embedding/base_distributed_embedding.py

```diff
@@ -262,31 +264,72 @@ def populate_placement_to_path_to_input_shape(
 
         super().build(input_shapes)
 
-    def call(
+    def preprocess(
         self,
         inputs: types.Nested[types.Tensor],
         weights: Optional[types.Nested[types.Tensor]] = None,
         training: bool = False,
     ) -> types.Nested[types.Tensor]:
-        """Lookup features in embedding tables and apply reduction.
+        """Preprocesses and reformats the data for consumption by the model.
+
+        This is necessary for the JAX backend with `sparsecore` placement in
+        order to enable `jit_compile=True` for the model.  In JAX, sparsecore
+        usage requires specially formatted data that depends on properties of
+        the available hardware.  This data reformatting currently does not
+        support jit-compilation, so must be applied _prior_ to feeding data into
+        a model.
+
+        An example usage might look like:
+        ```
+        # Create the embedding layer.
+        embedding_layer = DistributedEmbedding(feature_configs)
+
+        # Add preprocessing to a data input pipeline.
+        def training_dataset_generator():
+            for (inputs, weights), labels in iter(training_dataset):
+                yield embedding_layer.preprocess(
+                    inputs, weights, training=True
+                ), labels
+
+        preprocessed_training_dataset = training_dataset_generate()
+
+        # Construct, compile, and fit the model using the preprocessed data.
+        model = keras.Sequential(
+          [
+            embedding_layer,
+            keras.layers.Dense(2),
+            keras.layers.Dense(3),
+            keras.layers.Dense(4),
+          ]
+        )
+        model.compile(optimizer="adam", loss="mse", jit_compile=True)
+        model.fit(preprocessed_training_dataset, epochs=10)
+        ```
+
+        For non-JAX backends, preprocessing will bundle together the inputs and
```

Comment:
Can you explicitly mention that preprocessing is optional.

Reply:
Done.

Reply:
Added "optional" here again to re-iterate.